### PR TITLE
Make more `AuthAccount` functions `view`

### PR DESCRIPTION
**Pull request by dsainati1**
These functions were not marked as `view` when they could have been, causing unnecessary restrictive behavior. 

______

- [ ] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [X] Updated relevant documentation 
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Added appropriate labels

### CODE REVIEW

**Comment by joshuahannan** [comment]
Thank you! In my tests, I'm also seeing that `getCapability` is reported as impure, which should be safe too, right?

File: runtime/sema/authaccount.cdc [hunk 1]
```diff
@@ -83,7 +83,7 @@ access(all) struct AuthAccount {
     /// The given type must not necessarily be exactly the same as the type of the copied structure.
     ///
     /// The path must be a storage path, i.e., only the domain `storage` is allowed.
-    access(all) fun copy<T: AnyStruct>(from: StoragePath): T?
+    access(all) view fun copy<T: AnyStruct>(from: StoragePath): T?
```
Comment (SupunS) [inline runtime/sema/authaccount.cdc:86]: Can some of these other functions be `view` as well?
e.g: `borrow`, `check`,  `getCapability`, `getLinkTarget` - non of them change the state I guess

**Review by SupunS — APPROVED** [review]

File: runtime/sema/authaccount.cdc (hunk 1 above)
Comment (SupunS) [reply runtime/sema/authaccount.cdc:86]: lgtm!

**Review by turbolent — COMMENTED** [review]
Nice!

**Pull request merged**